src/daemon/utils_cache.c

```diff
@@ -649,6 +667,47 @@ int uc_get_value(metric_t const *m, value_t *ret) {
   return status;
 } /* value_t *uc_get_value */
 
+int uc_get_start_value_by_name(const char *name, value_t *ret_start_value,
+                               cdtime_t *ret_start_time) {
+  pthread_mutex_lock(&cache_lock);
+
+  cache_entry_t *ce = NULL;
+  int status = 0;
+  if (c_avl_get(cache_tree, name, (void *)&ce) == 0) {
+    assert(ce != NULL);
+
+    /* remove missing values from getval */
+    if (ce->state == STATE_MISSING) {
+      status = -1;
+    } else {
+      *ret_start_value = typed_value_clone(ce->start_value).value;
+      *ret_start_time = ce->start_time;
+    }
+  } else {
+    DEBUG("utils_cache: uc_get_start_value_by_name: No such value: %s", name);
+    status = -1;
+  }
```

Comment:
Consider inverting the logic here: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow